server/admin_management/api/app/routers/router_access_control_privilege.py

```diff
@@ -76,35 +76,32 @@ def create_access_control_privilege_many(
     )
 
     try:
-        audit_event_log.requesting_user = user_service.get_user_by_cognito_user_id(
-            requester.cognito_user_id
-        )
+        audit_event_log.requesting_user = requester
         audit_event_log.role = role_service.get_role_by_id(
             access_control_privilege_request.role_id
         )
         audit_event_log.application = audit_event_log.role.application
 
-        return access_control_privilege_service.create_access_control_privilege_many(
+        new_delegated_admin = access_control_privilege_service.create_access_control_privilege_many(
             access_control_privilege_request, requester.cognito_user_id, target_user
         )
+        # get target user from database, so for existing user, we'll have get the cognito user id
+        audit_event_log.target_user = user_service.get_user_by_domain_and_guid(
```

Comment:
The only difference is that, the target_user returns from router_guard won't have the cognito_user_id. So in the case the target_user is an existing user, by find user from database, we'll get the cognito_user_id.